### PR TITLE
BKR-1114: Added buildargs for Docker::Image::build invocation

### PR DESCRIPTION
**Pull request by barrymw**
Currently beaker acceptance tests will fail if executed behind a proxy.
By exposing the buildargs build parameter, it can be configured via ENV
DOCKER_BUILDARGS, or per host docker_buildargs

### CODE REVIEW

File: lib/beaker/hypervisor/docker.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ class Docker < Beaker::Hypervisor
     # Env variables supported:
     # DOCKER_REGISTRY: Docker registry URL
     # DOCKER_HOST: Remote docker host
+	# DOCKER_BUILDARGS: Docker buildargs map
```
Comment (ferglor) [inline lib/beaker/hypervisor/docker.rb]: fix indentation here

File: spec/beaker/hypervisor/docker_spec.rb [hunk 2]
```diff
@@ -162,7 +162,15 @@ module Beaker
...
-        expect( ::Docker::Image ).to receive(:build).with('special testing value', { :rm => true })
+        expect( ::Docker::Image ).to receive(:build).with('special testing value', { :rm => true, :buildargs => '{}' })
+
+        docker.provision
+      end
+
+      it 'should pass the buildargs from ENV DOCKER_BUILDARGS on to Docker::Image.create' do
+        allow( docker ).to receive(:dockerfile_for).and_return('special testing value')
+        ENV['DOCKER_BUILDARGS'] = 'HTTP_PROXY=http://1.1.1.1:3128'
+        expect( ::Docker::Image ).to receive(:build).with('special testing value', { :rm => true, :buildargs => "{\"HTTP_PROXY\":\"http://1.1.1.1:3128\"}" })
```
Comment (ferglor) [inline spec/beaker/hypervisor/docker_spec.rb]: can you add more tests for testing multiple buildargs?

Comment (barrymw) [inline spec/beaker/hypervisor/docker_spec.rb]: Added a secondary test and corrected the indentation.

**Comment by barrymw** [comment]
@kevpl I have updated the commit log.  Please review and let me know if its acceptable.  Thanks

**Comment by kevpl** [comment]
@barrymw sorry for the delay in response here. You comment is great, but what we're looking for is for your git commits to be changed.

The commits themselves seem fine, I don't think you need to squash them down, but they will need to have the ticket reference prefixed to each of them. You should be able to do this with an interactive rebase and picking 'edit' for each entry. Once that's done, we'll try to get this through the rest of our review process.

**Comment by barrymw** [comment]
Thanks for the pointers @kevpl, I have now corrected the commits and learnt some new git commands as well.

**Comment by ferglor** [comment]
👍

**Comment by kevpl** [comment]
@puppetlabs-jenkins test this please

**Comment by kevpl** [comment]
looks like a transient failure, unable to connect to a SUT in the middle of a test.

@puppetlabs-jenkins retest this please.

**Pull request merged**